### PR TITLE
refactor(ui): convert ipc to TypeScript

### PR DESCRIPTION
**Pull request by geigerzaehler**
We convert `native/ipc.js` to TypeScript. This enables us to use it from
TypeScript files (not only Svelte files) in `./ui`. The reason why we
need to convert the file to TypeScript is that rollup and typescript
don’t seem able to handle JS files and spit out a warning. In general it
also seems to be a good idea to move everything to TypeScript.

One result of this change is that we can remove everything Babel
related.

### CODE REVIEW

File: native/ipc.ts [hunk 1]
```diff
@@ -10,26 +13,27 @@ export const OPEN_PATH = "IPC_OPEN_PATH";
 //
 // The workaround is to use the electron native open dialog. As a bonus we
 // can configure it to allow users to create new directories.
-export const getDirectoryPath = () =>
+export const getDirectoryPath = (): void =>
```
Comment (rudolfs) [inline native/ipc.ts:16]:
```suggestion
export const getDirectoryPath = (): string =>
```

It returns the file path that is chosen within the file dialog.

Comment (geigerzaehler) [reply native/ipc.ts]: Fixed in https://github.com/radicle-dev/radicle-upstream/pull/1193/commits/79e2b4b5d2cab8a3bc660aab9f9e7ee8a4522c5f

**Review by xla — DISMISSED** [review]
🕹 🐢 🍀 🚊

**Review by rudolfs — DISMISSED** [review]
This was [blocking my issue](https://buildkite.com/monadic/radicle-upstream/builds/6196#5be0e621-8e7d-40a5-ade1-e6d6fb02c9ac/294-320).

Thanks for fixing it! 🙏

**Review by NunoAlexandre — DISMISSED** [review]
muy bueno!

**Review by rudolfs — APPROVED** [review]

**Review by NunoAlexandre — COMMENTED** [review]
:ping_pong:

**Review by NunoAlexandre — APPROVED** [review]

**Pull request merged**